core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -175,7 +175,14 @@ public ProcessingType getProcessingType() {
                                                                     processor, muleContext)))
                            .then(Mono.empty()))
         .onErrorResume(MessagingException.class,
-                       throwable -> Mono.from(event.getContext().error(throwable)).then(Mono.empty()))));
+                       throwable -> {
+                         if (!throwable.getEvent().getError().isPresent()) {
+                           // just in case anything in processor chain itself fails. We still have to create Error earlier on
+                           // though, so it's available in interceptors.
+                           throwable = updateMessagingException(processor).apply(throwable);
```

Comment:
Will do, when build is green :-( Anyhow this only covers a very specific scenario where updating the payload for streaming within the chain interceptors themselves result in a NPE, otherwise error should never be null here.